### PR TITLE
Form styles

### PR DESCRIPTION
**Pull request by lal65**
# Description:
Implement the most basic element styles for various form related things.
- [x] Details
- [x] Fieldsets
- [x] Forms
- [x] Checkbox inputs
- [x] Color inputs
- [x] Date inputs
- [x] Text inputs
- [x] Email inputs
- [x] File inputs
- [x] Number inputs
- [x] password inputs
- [x] Radio inputs
- [x] Range inputs
- [x] Search inputs
- [x] Phone inputs
- [x] Time inputs
- [x] URL inputs
- [x] labels
- [x] legends
- [x] Selects
- [x] Textareas

New auto-contrasting on existing things.
- [x] 1. Add auto-contrasting to paragraphs and links.
- [x] 2. Add auto-contrasting to alerts.
- [x] 3. Add auto-contrasting to intro forms
image

Add an "info" flavor alert.
image

Add a boatload of examples.

## Metadata
Delete sections that are not relevant.

### Workfront Task Link
  https://pennstateoutreach.my.workfront.com/task/613118040004631d171a19dd2cc5e114/overview
### Adobe XD Link
  https://xd.adobe.com/view/c3926599-fc79-48ce-9ab2-cb4e483bfc33-96c1/
### Review environment Link
  https://developers.outreach.psu.edu/form-styles

### CODE REVIEW

**Review by lal65 — COMMENTED** [review]
Notes - down to `input-checkbox`.

File: packages/details/src/scss/_summary.scss [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    content: '';
+    -webkit-mask-size: _rem(2.4) auto;
+    mask-size: _rem(2.4) auto;
+    -webkit-mask: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M233 407c13 12 33 12 46 0l192-192a32 32 0 0 0-46-46L256 339 87 169a32 32 0 0 0-46 46l192 192z"/></svg>') no-repeat center;
+    mask: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M233 407c13 12 33 12 46 0l192-192a32 32 0 0 0-46-46L256 339 87 169a32 32 0 0 0-46 46l192 192z"/></svg>') no-repeat center;
+    transition: transform var(--transition-duration--default) linear,
+    background-color var(--transition-duration--default) linear;
+  }
+
+  &.form-required::after {
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/details/src/scss/_summary.scss]: Moved to integration layer.

File: packages/input-checkbox/src/scss/styles.scss [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+      &:checked:hover::after {
+        background-color: var(--pa-link);
+      }
+    }
+
+    &:disabled {
+      &::before {
+        background-color: var(--form-element-disabled-background-color);
+        color: var(--white);
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/input-checkbox/src/scss/styles.scss]: Axed!

File: packages/details/src/scss/_summary.scss [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    font-size: inherit;
+  }
+
+  // Hide empty summaries.
+  > .summary:empty {
+    display: none;
+  }
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/details/src/scss/_summary.scss]: Moved to integration layer.

File: packages/select/src/scss/styles.scss [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      box-shadow: none;
+    }
+  }
+}
+
+.form-select-wrapper:has(select.error) {
+  background-color: var(--error-red);
+  border-top-left-radius: var(--form-element-border-radius--rounded);
+  border-top-right-radius: var(--form-element-border-radius--rounded);
+}
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/select/src/scss/styles.scss]: Moved to integration layer

File: packages/select/src/scss/styles.scss [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    background-repeat: no-repeat;
+    background-position: calc(100% - var(--form-element-padding--horizontal)) center;
+    background-size: auto _rem(.8);
+  }
+
+  &.error {
+    border: _rem(.1) solid var(--error-red);
+  }
+
+  &:not([multiple]) {
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/select/src/scss/styles.scss]: Combined.

File: packages/input-radio/src/scss/styles.scss [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+          border-color: var(--light-mauve);
+        }
+
+        &:checked::after {
+          background-color: var(--slate);
+        }
+
+        & ~ * {
+          color: var(--medium-grey);
+        }
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/input-radio/src/scss/styles.scss]: Removed.

File: packages/input-time/src/scss/styles.scss [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+input[type="time"] {
+  // Android, in particular, renders native time inputs with a chevron versus
+  // a more intuitive clock icon. To normalize this, we add our own icon via
+  // a background image.
+  appearance: none;
+  $icon-width: #{_rem(2.4)};
+  background: var(--form-element-background-color) url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 24 24"><path fill="%23001e44" d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"/><path d="M0 0h24v24H0z" fill="none"/><path fill="%23001e44" d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"/></svg>') no-repeat calc(100% - var(--form-element-padding--horizontal)) center / $icon-width;
+  padding-right: calc($icon-width + var(--form-element-padding--horizontal) * 2);
+  transition: background-size var(--transition-duration--default) linear;
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/input-time/src/scss/styles.scss]: Axed.

File: packages/input-date/src/scss/styles.scss [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+input[type="date"] {
+  // Android, in particular, renders native date inputs with a chevron versus
+  // a more intuitive calendar icon. To normalize this, we add our own icon via
+  // a background image.
+  appearance: none;
+  $icon-width: #{_rem(2.4)};
+  background: var(--form-element-background-color) url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" width="16" height="15" viewBox="0 0 24 24"><path fill="%23001e44" d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"/><path fill="none" d="M0 0h24v24H0z"/></svg>') no-repeat calc(100% - var(--form-element-padding--horizontal)) center / $icon-width;
+  padding-right: calc($icon-width + var(--form-element-padding--horizontal) * 2);
+  transition: background-size var(--transition-duration--default) linear;
```
[the remark this answers is not in the document]
Comment (lal65) [reply packages/input-date/src/scss/styles.scss]: Nuked.

File: packages/base/src/scss/_tables.scss [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+      width: 100%;
+      display: block;
+      border: none;
+    }
+
+    td {
+      border: none;
+    }
+
+    &.show td {}
```
Comment (kjl16) [inline packages/base/src/scss/_tables.scss:105]: Can delete this

File: packages/base/src/scss/_tables.scss [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+      color: var(--beaver-blue);
+    }
+
+    p:last-child {
+      margin-bottom: 0;
+    }
+
+    &.st-head-row:not(.st-head-row-main) {
+      &, & p {
+        font-size: var(font-size--small);
```
Comment (kjl16) [inline packages/base/src/scss/_tables.scss:55]: --font-size--small

**Review by kjl16 — CHANGES_REQUESTED** [review]
Just a couple small things

**Review by kjl16 — APPROVED** [review]

**Pull request merged**